lib/obs_github_deployments/cli/commands/set_lock.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module ObsGithubDeployments
+  module CLI
+    module Commands
+      class SetLock < Dry::CLI::Command
+        desc "Lock deployments for a specific GitHub repository"
+
+        option :repository, default: ENV["GITHUB_REPOSITORY"],
+                            desc: "GitHub repository name where deployments should get locked"
+        option :ref, default: ENV["GITHUB_BRANCH"],
+                     desc: "GitHub branch name the locked deployment is referring to"
+        option :token, default: ENV["GITHUB_TOKEN"],
+                       desc: "GitHub authentication token used to authenticate against the API"
+        option :reason, required: true, desc: "Explain reasoning behind locking the deployment process"
+
+        def call(**options)
+          abort("You need to provide a repository name") unless options[:repository]
+          abort("You need to provide a branch name") unless options[:ref]
+          abort("You need to provide a token in order to authenticate") unless options[:token]
+
+          begin
+            ObsGithubDeployments::Deployment.new(repository: options[:repository],
+                                                 access_token: options[:token],
+                                                 ref: options[:ref]).lock(reason: options[:reason])
+          rescue ObsGithubDeployments::Deployment::PendingError,
+                 ObsGithubDeployments::Deployment::AlreadyLockedError,
+                 ObsGithubDeployments::Deployment::NoReasonGivenError => e
+            abort(e.message)
+          end
+          exit(0)
```

Comment:
```suggestion
          ObsGithubDeployments::Deployment.new(repository: options[:repository],
                                               access_token: options[:token],
                                               ref: options[:ref]).lock(reason: options[:reason]) 
```

Reply:
exit(0) is, IMHO, not needed.

What is being done inside the `rescue` makes no difference with throwing a normal error. Do we explicitly want only these exceptions to make the program abort? In other words, don't we want other possible exceptions to make the program NOT to abort?

Reply:
At this point I only wanted to rescue exception I know of, for sure it would be good to also check what possible exceptions we might get through Octokit and to also handle them. The unhandled exception will still abort the programm, no?

Reply:
Citing myself: "Do we explicitly want only these exceptions to make the program abort?"

Yes, we want to have a clearer error description in those cases. The normal error stack is not needed.

> The unhandled exception will still abort the programm, no?

Yes. Sorry for the confusion.